---
 tasks/task_save.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tasks/task_save.c b/tasks/task_save.c
index 08bcf1f67f..957a1d3943 100644
--- a/tasks/task_save.c
+++ b/tasks/task_save.c
@@ -62,8 +62,8 @@
 #include "../cheat_manager.h"
 #endif
 
-#if defined(HAVE_LIBNX) || defined(_3DS)
-#define SAVE_STATE_CHUNK 4096 * 10
+#if defined(HAVE_LIBNX) || defined(_3DS) || defined(EMSCRIPTEN)
+#define SAVE_STATE_CHUNK 4096 * 100
 #else
 #define SAVE_STATE_CHUNK 4096
 #endif
-- 

